thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/dashboard/resources/EmailResource.java

```diff
@@ -81,6 +69,74 @@ public Response deleteByAlertId(@PathParam("alertId") Long alertId) {
     return Response.ok().build();
   }
 
+  /**
+   * update alert config's cron and activation by id
+   * @param id alert config id
+   * @param cron cron expression for alert
+   * @param isActive activate or not
+   * @return Response
+   * @throws Exception
+   */
+  @PUT
+  @Path("/alert/{id}")
+  public Response updateAlertConfig(@NotNull @PathParam("id") Long id,
+      @QueryParam("cron") String cron, @QueryParam("isActive") Boolean isActive) throws Exception {
+
+    AlertConfigDTO alert = alertDAO.findById(id);
+    if (alert == null) {
+      throw new IllegalStateException("Alert Config with id " + id + " does not exist");
+    }
+
+    if (isActive != null) {
+      alert.setActive(isActive);
+    }
+
+    if (StringUtils.isNotEmpty(cron)) {
+      // validate cron
+      if (!CronExpression.isValidExpression(cron)) {
+        throw new IllegalArgumentException("Invalid cron expression for cron : " + cron);
+      }
+      alert.setCronExpression(cron);
+    }
+
+    alertDAO.update(alert);
+    return Response.ok(id).build();
+  }
+
+  /**
+   * update alert config's holiday cron and activation by id
+   * if this cron is not null then holiday mode is activate
+   * @param id id of the config
+   * @param cron holiday cron expression
+   * @return Response
+   * @throws Exception
+   */
+  @PUT
+  @Path("/alert/{id}/holiday-mode")
+  public Response updateAlertConfigHolidayCron(@NotNull @PathParam("id") Long id,
+      @QueryParam("cron") String cron) throws Exception {
```

Comment:
thanks for implementing. please move this to AlertJobResource.

Reply:
AlertJobResource is being deprecated. and We should not add more functions into it.